### PR TITLE
updated --dev flag to --all-dev

### PR DESCRIPTION
**Pull request by erinxocon**
Changed name of the --dev flag to --all-dev to be more specific about what it does.  --dev now does nothing

### CODE REVIEW

**Comment by erinxocon** [comment]
This address #837

File: tests/test_pipenv.py [hunk 1]
```diff
@@ -271,6 +271,28 @@ def test_uninstall_all_local_files(self):
             assert 'tablib' in c.out
             assert 'tablib' not in p.pipfile['packages']
 
+    @pytest.mark.run
+    @pytest.mark.uninstall
+    def test_uninstall_all_dev(self):
+        with PipenvInstance() as p:
+            c = p.pipenv('install --dev requests pytest')
```
Comment (techalchemy) [inline tests/test_pipenv.py]: Not sure if you're still working on this and if I'm just prematurely reviewing but you may want to install some non-dev stuff and test that it isn't getting removed/deleted

Comment (erinxocon) [reply tests/test_pipenv.py]: Good thought

**Comment by erinxocon** [comment]
@techalchemy This can be reverted easily.

My thinking was that since the perceived default behavior of --dev is not what it actually does.  The error message is nice, but then I feel like most people will select No from that prompt and be frustrated as to why it didn't work.  I'm not sure though, UX isn't my fotré.  I struggled too with the behavior of --all and --all-dev being different in terms of scrubbing the Pipfile, but I don't know what to do there...

**Comment by nateprewitt** [comment]
Errm, this is a breaking change and we'd need to do a major release for our next release now.

**Comment by nateprewitt** [comment]
I see we've already released it :/ Well... hmm. I guess we wait and see if there's outcry.

**Comment by erinxocon** [comment]
Sorry about that.   I take responsibility.  I need to incorporate these ideas before pressing merge.

**Pull request merged**